src/raiden_libs/blockchain.py

```diff
@@ -296,3 +309,71 @@ def get_pessimistic_udc_balance(
         udc.functions.effectiveBalance(address).call(block_identifier=BlockNumber(block))
         for block in range(from_block, to_block + 1)
     )
+
+
+def get_blockchain_events_adaptive(
```

Comment:
This somewhat duplicates what we have in Raiden, but I guess extracting just the reusable part is not trivial.

Reply:
This is tracked in https://github.com/raiden-network/raiden-services/issues/672